### PR TITLE
added first solution

### PR DESCRIPTION
**Pull request by IhorRome**

### CODE REVIEW

**Review by andriiromaniuk — APPROVED** [review]

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -3,48 +3,146 @@
...
+        final Node<T> currentTail = last;
+        final Node<T> newNode = new Node<>(last, value, null);
+        last = newNode;
+        if (currentTail == null) {
+            first = newNode;
+        } else {
+            currentTail.next = newNode;
+        }
+        size++;
+        return true;
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java:34]:
```suggestion
        final Node<T> newNode = new Node<>(last, value, null);
        if (last == null) {
            first = newNode;
        } else {
            last.next = newNode;
        }
        last = newNode;
        size++;
        return true;
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -3,48 +3,146 @@
...
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> lostNode = first;
+        while (lostNode != null) {
+            if (lostNode.item == object || lostNode.item != null && lostNode.item.equals(object)) {
+                unlink(lostNode);
+                return true;
+            } else {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java:88]: do we need else block here?

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -3,48 +3,146 @@
...
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+        } else {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/MyLinkedList.java:41]: we can add return after add() method and remove else block

**Review by Sofasmile — CHANGES_REQUESTED** [review]
good job)

**Review by Sofasmile — APPROVED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**